### PR TITLE
fix(windows): bun `link`, `unlink`, and fix tests in `test/cli/install`

### PR DESCRIPTION
**Pull request by dylan-conway**
### What does this PR do?
gets bun `link` and `unlink` working on windows.

also fixes small bugs on windows for:
- `toMatchSnapshot`
- `bun pm ls`
- `bun pm cache rm`
- `bun create`

### How did you verify your code works?
ci and manually

### CODE REVIEW

File: src/cli/bunx_command.zig [hunk 1]
```diff
@@ -377,45 +377,48 @@ pub const BunxCommand = struct {
...
-
-                    // Only use the system-installed version if there is no version specified
-                    if (update_request.version.literal.isEmpty()) {
-                        destination_ = bun.which(
+            const root_dir_fd = root_dir_info.getFileDescriptor();
+            if (root_dir_fd != .zero) {
+                if (getBinName(&this_bundler, root_dir_fd, bunx_cache_dir, initial_bin_name)) |package_name_for_bin| {
+                    // if we check the bin name and its actually the same, we don't need to check $PATH here again
+                    if (!strings.eqlLong(package_name_for_bin, initial_bin_name, true)) {
+                        absolute_in_cache_dir = std.fmt.bufPrint(&absolute_in_cache_dir_buf, "{s}/node_modules/.bin/{s}", .{ bunx_cache_dir, package_name_for_bin }) catch unreachable;
```
Comment (Jarred-Sumner) [inline src/cli/bunx_command.zig:385]: is the windows path join going to be an issue?

File: src/cli/package_manager_command.zig [hunk 2]
```diff
@@ -179,13 +179,16 @@ pub const PackageManagerCommand = struct {
...
             var dir: [bun.MAX_PATH_BYTES]u8 = undefined;
-            const fd = pm.getCacheDirectory();
-            const outpath = bun.getFdPath(fd.fd, &dir) catch |err| {
+            var fd = pm.getCacheDirectory();
+            var outpath = bun.getFdPath(fd.fd, &dir) catch |err| {
                 Output.prettyErrorln("{s} getting cache directory", .{@errorName(err)});
                 Global.crash();
             };
 
+            outpath = Path.normalizeString(outpath, true, .auto);
```
Comment (Jarred-Sumner) [inline src/cli/package_manager_command.zig]: why `getFdPath` and then normalize?

Comment (dylan-conway) [reply src/cli/package_manager_command.zig]: `getFdPath` always has forward slashes

Comment (dylan-conway) [reply src/cli/package_manager_command.zig]: updated

**Pull request merged**